### PR TITLE
Separate GitHub workflow jobs

### PR DESCRIPTION
**Pull request by sealesj**
Create prerequisite job for python setup and deps_parser in order to avoid having custom/unsupported actions in the same job as the scorecards_analysis

In order to make the setup job a prerequisite, I used the keyword "needs" for the scorecards job and then linked the setup job

Context: https://github.com/ossf/scorecard-action/issues/910

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: .github/workflows/scorecards-analysis.yml [hunk 1]
```diff
@@ -9,30 +9,36 @@ on:
 permissions: read-all
 
 jobs:
-  analysis:
-    name: Scorecards analysis
+  setup:
+    name: Scorecards analysis setup
```
Comment (drewroengoogle) [inline .github/workflows/scorecards-analysis.yml:13]: Question: Is this piece related to scorecards? It looks like the script parses the DEPS file but does scorecards utilize that parsed file?

**Review by drewroengoogle — APPROVED** [review]
Approved with question

File: .github/workflows/scorecards-analysis.yml (hunk 1 above)
Comment (sealesj) [reply .github/workflows/scorecards-analysis.yml:13]: It does parse the DEPS file and creates the flattened dependency file that I am using in vulnerability scanning. 
The deps scanning action has already been running in preparation for the vulnerability scanning which will happen in the steps afterwards. 
I'm not changing any of that with these changes, just refactoring

Comment (drewroengoogle) [reply .github/workflows/scorecards-analysis.yml:13]: That makes sense. My question was pointed toward whether this belongs in the scorecards action at all or if it would better fit in its own vulnerability scanning action.

Comment (sealesj) [reply .github/workflows/scorecards-analysis.yml:13]: Ah ok I understand. I suppose we could make a new job altogether that for right now has just this deps_parser step in it. You are right, it isn't necessarily related to scorecards, it is setup step for the later vuln scanning
